docs/rfcs/internal-telemetry.md

```diff
@@ -0,0 +1,66 @@
+# Defining guidelines for internal telemetry
+
+## Overview
+
+The Collector supports generating internal telemetry
+that assists end users when operating the Collector. So
+far, much of the telemetry has been added to components as
+needed, without guidelines for component authors to provide
+a consistent experience to end users. The goal of this document
+is to:
+
+- describe the naming and attributes currently in use
+- define consistent units + naming for telemetry emitted by the Collector
+- define the process that should be used to configure new metrics
+
+## Out of scope
+
+This document is not intending to dictate when telemetry should be
+emitted by various Collector components. Considering the various types
+of components, this will be better discussed in a future document.
+
+This document is not intending to provide a comprehensive plan for how
+the Collector or the health of telemetry pipelines should be monitored. There
+is an [OpenTelemetry Enhancement Proposal](https://github.com/open-telemetry/oteps/pull/259) that has already started the process
+to provide this information. The information in this document will be updated once
+the OTEP lands to follow its recommendations.
+
+## Internal telemetry properties
+
+Telemetry produced by the Collector have the following properties:
+
+- metrics produced by Collector components use the prefix `otelcol_`. Until v0.106.0, this
+  was done via the Prometheus namespace, which made metrics inconsistent for users wishing
+  to emit metrics via another exporter. See https://github.com/open-telemetry/opentelemetry-collector/issues/9315
+  for more details
```

Comment:
The v0.106.0 is (or will quickly become) a historical detail, so maybe we can just not mention it here

Reply:
removed